### DISCUSSION TITLE
What command to use for "Deploying the local code" vs "deploy from a remote repo"

### DISCUSSION THREAD

**Discussion by rberrelleza** [okteto/okteto#2462] [Ideas]
(this conversation started in #2456, moving it here since it might impact more than one PR). 

I think that "Deploying the local code" vs "deploy what I have in a remote repo" are very different actions and that a developer is going to have a different motivation and workflow when doing one vs the other. 

Today, we are proposing using `okteto deploy` and `okteto destroy` for both scenarios. For local code, the user would use `okteto deploy`. To deploy what you have on a remote repo/branch, you'd use `okteto deploy --repo XX --branch XXX`. I don't think that is going to be obvious to the user that passing the branch means that it'll run on cloud vs local. It's like we are merging two commands into one.

A few proposals from @rlamana and @rberrelleza:
1.  Add a  --remote option to `okteto deploy`
2. Create a 'remote' subcommand, so that, to deploy a repo on github, you'll use `okteto remote deploy --repo X --branch`. 

Any other ideas?

**Comment by rberrelleza** [comment]
One thing to consider here is that most commands on Okteto are 'remote' , but use the local code.  a 'deploy from github' is the different one.

**Reply by jLopezbarb** [reply]
I think there are two reasons why it is better to deploy repositories remotely when adding repository or branch flags:
- When you launch `Okteto Pipeline deploy` there is a message which tells you that the command is going to be deprecated in favor of `okteto deploy`, which as a user I assume I will be able to deploy my repositories in the same way as before, i.e. if I launch the application with the `repository` flag I expect it to deploy that repository and not the one I have in the current local directory.
- Adding the `remote` flag (and deprecating the `repository` and `branch` flags) I would need to clone the repo and get inside the folder to be able to launch that new repository, which I think makes the repository deployment more complex.

We could add a new subcommand for deploying on remote but we will have to fight again with commands working on okteto vs commands that runs on every cluster

**Reply by jLopezbarb** [reply]
for the okteto destroy command as we don't need the `branch` and `repository` flags (only the `name`), we could add the `remote` flag and it will launch as it was on okteto CLI version 1.X.X

**Reply by rberrelleza** [reply]
Let's not worry about the past and deprecated commands for now, and focus on how to give the best experience to the users of okteto cli. Once we land that, we can figure out how to communicate it back and all that.

**Reply by rberrelleza** [reply]
@pchico83 @jLopezbarb can we remove the deprecation on okteto preview deploy on v2 for now, while you work on a spec for remote deployments?

**Comment by pchico83** [comment]
I don't have full clarity on this yet, but if `okteto deploy` somehow runs on remote (for example by synching the code in a remote container) both operations could be unified.

**Reply by rberrelleza** [reply]
I think even in that case we still have two different use cases:
1. deploy the code I'm currently working on 
2. deploy the code that is committed on a remote branch

I still believe that we should have two different commands, rather than just a repo. 

@pchico83 what is the use case for the user to run the  "deploy the code that is committed on a remote branch" command? Also, is this something that will work in all clusters, or only on Okteto clusters?

**Reply by pchico83** [reply]
It should work on any cluster. My feeling is that `2) deploy the code that is committed on a remote branch` is covered by `okteto preview deploy`. But I need to work on a full spec for this and analyze all the use cases.
